gender_novels/analysis/analysis.py

```diff
@@ -202,6 +204,180 @@ def instance_dist(novel, word):
                 count = 0
     return output
 
+
+def male_instance_dist(text):
+    """
+        Takes in a novel, returns a list of distances between each instance of a male pronoun in that novel
+       >>> from gender_novels import novel
+       >>> summary = "James was his convicted of adultery. "
+       >>> summary += "which made him very sad, and then his Jane was also sad, and himself everybody was "
+       >>> summary += "sad and then he died and it was very sad. His Sadness."
+       >>> novel_metadata = {'author': 'Hawthorne, Nathaniel', 'title': 'Scarlet Letter',
+       ...                   'corpus_name': 'sample_novels', 'date': '1966',
+       ...                   'filename': None, 'text': summary}
+       >>> scarlett = novel.Novel(novel_metadata)
+       >>> male_instance_dist(scarlett)
+       [6, 5, 6, 6, 7]
+
+       :param:tokenized text of a novel
+       :return: list of distances between instances of gendered word
+
+       """
+    output = []
+    count = 0
+    start = False
+    words = ["his", "him", "he", "himself"]
+
+    for e in text:
+        e = e.lower()
+        if not start:
+            if e in words:
+                start = True
+        else:
+            count += 1
+            if e in words:
+                output.append(count)
+                count = 0
+    return output
+
+
+def female_instance_dist(text):
+    """
+        Takes in a novel, returns a list of distances between each instance of a female pronoun in that novel
+       >>> from gender_novels import novel
+       >>> summary = "James was her convicted of adultery. "
+       >>> summary += "which made her very sad, and then she Arthur was also sad, and herself everybody was "
+       >>> summary += "sad and then she died and it was very sad. Her Sadness."
+       >>> novel_metadata = {'author': 'Hawthorne, Nathaniel', 'title': 'Scarlet Letter',
+       ...                   'corpus_name': 'sample_novels', 'date': '1966',
+       ...                   'filename': None, 'text': summary}
+       >>> scarlett = novel.Novel(novel_metadata)
+       >>> female_instance_dist(scarlett)
+       [6, 5, 6, 6, 7]
+
+       :param:tokenized text of a novel
+       :return: list of distances between instances of gendered word
+
+       """
+    output = []
+    count = 0
+    start = False
+    words = ["her", "hers", "she", "herself"]
+
+    for e in text:
+        e = e.lower()
+        if not start:
+            if e in words:
+                start = True
+        else:
+            count += 1
+            if e in words:
+                output.append(count)
+                count = 0
+    return output
+
+
+def find_male_adj(novel):
+    """
```

Comment:
Hi Myke,
Thanks for the feedback! I think I've made all the changes that you suggested. One possible problem that I ran into while testing it on some novels was that I think sometimes nltk misclassifies adjectives as nouns/adverbs and vice-versa. I'm not entirely sure why this is, it may be a problem within nltk itself or possibly an issue that stems from considering the 'windows' of words rather than full sentences.

Reply:
"The orange carrot" and "The ripe orange" -- in isolation, the word "orange" can't be properly classified.  So this is why @asselism and @varmichelle have been working on the sentence+ level classifiers.  It's not something that we can fix in this method.  We always need to be aware of the limitations of our analytical method, disclose them in our writeups, and then decide if those limitations are strong enough to prevent publishing or not.  I suspect here, it'll be rare enough that we can generally go ahead.